File: java/java/src/main/java/org/signal/zkgroup/receipts/ReceiptSerial.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.receipts;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;

public final class ReceiptSerial extends ByteArray {

  public static final int SIZE = 16;

  public ReceiptSerial(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

}

File: node/ts/zkgroup/receipts/ReceiptSerial.ts
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray.js';

export default class ReceiptSerial extends ByteArray {
  private readonly __type?: never;
  static SIZE = 16;

  constructor(contents: Uint8Array) {
    super(contents, ReceiptSerial.checkLength(ReceiptSerial.SIZE));
  }
}
